cmd/cli/update.go

```diff
@@ -0,0 +1,142 @@
+			}
+			upd.clientSet = clientset
+			return upd.run(config)
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVar(&upd.cfg.containerRegistry, "container-registry", "openservicemesh", "container registry that hosts control plane component images")
+	f.StringVar(&upd.cfg.osmImageTag, "osm-image-tag", "v0.2.0", "osm image tag")
+	f.StringVar(&upd.cfg.containerRegistrySecret, "container-registry-secret", "acr-creds", "name of Kubernetes secret for container registry credentials to be created if it doesn't already exist")
+	f.StringVar(&upd.cfg.chartPath, "osm-chart-path", "", "path to osm chart to override default chart")
+	f.StringVar(&upd.cfg.certManager, "certificate-manager", defaultCertManager, "certificate manager to use (tresor or vault)")
+	f.StringVar(&upd.cfg.vaultHost, "vault-host", "", "Hashicorp Vault host/service - where Vault is installed")
+	f.StringVar(&upd.cfg.vaultProtocol, "vault-protocol", defaultVaultProtocol, "protocol to use to connect to Vault")
+	f.StringVar(&upd.cfg.vaultToken, "vault-token", "", "token that should be used to connect to Vault")
+	f.StringVar(&upd.cfg.vaultRole, "vault-role", "openservicemesh", "Vault role to be used by Open Service Mesh")
+	f.IntVar(&upd.cfg.serviceCertValidityMinutes, "service-cert-validity-minutes", defaultCertValidityMinutes, "Certificate TTL in minutes")
+	f.StringVar(&upd.cfg.prometheusRetentionTime, "prometheus-retention-time", constants.PrometheusDefaultRetentionTime, "Duration for which data will be retained in prometheus")
+	f.BoolVar(&upd.cfg.enableDebugServer, "enable-debug-server", false, "Enable the debug HTTP server")
+	f.BoolVar(&upd.cfg.enablePermissiveTrafficPolicy, "enable-permissive-traffic-policy", false, "Enable permissive traffic policy mode")
+	f.BoolVar(&upd.cfg.enableEgress, "enable-egress", false, "Enable egress in the mesh")
+	f.StringSliceVar(&upd.cfg.meshCIDRRanges, "mesh-cidr", []string{}, "mesh CIDR range, accepts multiple CIDRs, required if enable-egress option is true")
+	f.BoolVar(&upd.cfg.enableBackpressureExperimental, "enable-backpressure-experimental", false, "Enable experimental backpressure feature")
+	f.BoolVar(&upd.cfg.enableMetricsStack, "enable-metrics-stack", true, "Enable metrics (Prometheus and Grafana) deployment")
+	f.StringVar(&upd.cfg.meshName, "mesh-name", defaultMeshName, "name for the new control plane instance")
+	f.BoolVar(&upd.cfg.deployZipkin, "deploy-zipkin", true, "Deploy Zipkin in the namespace of the OSM controller")
+
+	return cmd
+}
+
+func (u *updateCmd) run(config *helm.Configuration) error {
+	var chartRequested *chart.Chart
+	var err error
+	if u.cfg.chartPath != "" {
+		chartRequested, err = loader.Load(u.cfg.chartPath)
+	} else {
+		chartRequested, err = cli.LoadChart(chartTGZSource)
+	}
+	if err != nil {
+		return err
+	}
+
+	values, err := u.cfg.resolveValues()
+	if err != nil {
+		return err
+	}
+
+	// TODO (nitishm): Should we check if the controlplane deployment exists ?
```

Comment:
@michelleN Do we want to check for existing deployments to perform the upgrade. My hunch is that it is irrelevant but I would rather discuss this first.